### ISSUE TITLE
JSON type should be supported for replication and saves.

### ISSUE THREAD

**Issue by boazsade** [dragonflydb/dragonfly#587]
**Did you search GitHub Issues and GitHub Discussions First?**
This become an issue after #238 that added the new type JSON.

**Is your feature request related to a problem? Please describe.**
This can become an issue in case a save or replication is made when the dataset contain JSON type.

**Describe the solution you'd like**
Being able to save/replication dataset that contain JSON entries.
For example:
Running the following
```
JSON.SET doc $ '{"a":2}'
```
will add a new entry for key "doc" with the value type being JSON.
This then cannot be replication or save since this type is not part of the RDB type system.
It is not expected that this would become part of RDB types, and would be compatible with REDIS.

**Describe alternatives you've considered**
Since the JSON type is not part of the RDB data types, the only way would be to save this as a string and somehow know that it should then read back as JSON, but this would have the issue that this is not supported as well.

**Additional context**
This would required it own formatting as JSON has more internal types (such as bool, float, string, arrays, decimal and so on) than RDB, and it is not either subset or super set of it.

**Comment by romange** [comment]
Also for dump, restore
